codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java

```diff
@@ -1550,6 +1556,27 @@ public HttpHeaders set(HttpHeaders headers) {
     }
 
     /**
+     * Retains all current headers but calls {@link #set(String, Object)} for each entry in {@code headers}
+     *
+     * @param headers The headers used to {@link #set(String, Object)} values in this instance
+     * @return {@code this}
+     */
+    public HttpHeaders setAll(HttpHeaders headers) {
+        if (headers == null) {
+            throw new NullPointerException("headers");
+        }
+
+        if (headers.isEmpty()) {
```

Comment:
Consider staying consistent with [DefaultHeaders.setAll(..)](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L743).  This involves using the `forEachEntry` construct.